a10_octavia/controller/worker/tasks/a10_network_tasks.py

```diff
@@ -658,3 +661,44 @@ def revert(self, result, loadbalancer, amp_data=None, update_dict=None,
         except Exception as e:
             LOG.error('Failed to remove QoS policy: %s from port: %s due '
                       'to error: %s', orig_qos_id, amp_data.vrrp_port_id, e)
+
+
+class HandleVRRPFloatingIPDelta(BaseNetworkTask):
+    """Handle VRRP Floating IP port delta"""
+
+    @axapi_client_decorator
+    def execute(self, vthunder, member):
```

Comment:
```python
@axapi_client_decorator
def execute(self, vthunder, member):
    fip_port = None
    floating_ip = vthunder.vrrp_floating_ip

    if floating_ip:
        if floating_ip == 'dhcp':
            fip_port = self.network_driver.create_port(member.subnet_id)
        else:
            subnet_ip, subnet_mask = get_net_info_from_subnet(subnet_id)
            if not a10_utils.check_ip_in_subnet_range(floating_ip, subnet_mask):
                LOG.exception("Invalid VRID floating IP. IP out of subnet range: %s", floating_ip)
                raise

            vrid = self.axapi_client.vrrpa.vrid.get(0)
            vrid_ip = vrid['floating-ip']['ip-address-cfg'][0]['ip-address']
            if floating_ip != vrid_ip:
                fip_port = self.network_driver.create_port(member.subnet_id, fixed_ip=floating_ip)

        if fip_port:
            self.axapi_client.vrrpa.vrid.update(0, fip_port.fixed_ip.ip_address)

    if vthunder.vrrp_port_id:
        self.network_driver.delete_port(vthunder.vrrp_port_id)

    return fip_port
```

Reply:
This fails in following test case:
if I am updating a member, without any change in it (floating_ip == vrid_ip)
1. New fip_port wont get created
2. The older port will get deleted as 'vthunder.vrrp_port_id' is not None 

I think the change would be, 
if vthunder.vrrp_port_id and fip_port:
        self.network_driver.delete_port(vthunder.vrrp_port_id)

Reply:
Case 1.1: If floating_ip == vrid_ip we don't want to create a new port.
Case 1.2: Good catch. However, we want to handle the case in which no floating_ip is defined in the config yet there is a vrrp_port. We can check `vthunder.vrrp_port_id and floating_ip`

Case 2: If this happens, then the operator didn't configure things properly ahead of time as agreed upon and we should 100% fail. We can't handle everything like we do with standard amphora.

Reply:
there are many bugs with this code, we need to resolve them in pair programming. I am not sure why we are refactoring code, if previous one is covering most of the UTs. Adding remaining test cases is much easier in older code than this refactored one

Reply:
I'm trying to refactor this code as:
1. It doesn't handle all cases
2. Creates variables that aren't used where they should be (see 'ip_address') with poor naming
3. Is hard to read due to the over use of if statements and returns
4. Doesn't actually update the floating ip on the box itself

All this does in the current state is create or delete a port.

Yes we can devote some time to pair programming to finish this.